Bash script to configure files

samratjha96 opened the pull request:
A bash script to walk the user through all the files configuration that needs to be done before the pipeline can be kicked off

Commit:
Bash script to configure files

samratjha96 commented:
@cobylu @hankjtaylor Ready for CR! Probably gonna be hard to read bash so best to just run it on your local fork of the quickstart repo and look for problems there. Just pay attention to all the prompts, messages for any improvements. 

And I have marked the script with a couple of TODO's. Let me know if you guys think they are worth pursuing

samratjha96 commented on configurePipelineFiles.sh:
```diff
@@ -0,0 +1,226 @@
+
+  echo -e "\\n${BLACK_ON_GREY}Modifying $VERSION_MANAGER_PROPERTIES with the following prompts${RESET}\\n"
+  read -p "$(echo -e $WHITE)Enter your repo URL (repo where you want to version your applications):$(echo -e $RESET) " REPO_URL
+  read -p "$(echo -e $WHITE)Enter a folder name where you want this repo to be cloned (ex: staging):$(echo -e $RESET) " LOCAL_REPO
```
We hardcode this in the Jenkinsfile in one of your recent commits right? Should I just take away this line?

hankjtaylor replied:
Yep! you can take this out

samratjha96 commented on configurePipelineFiles.sh:
```diff
@@ -0,0 +1,226 @@
+  echo -e "\\n${BLACK_ON_GREY}Modifying $VERSION_MANAGER_PROPERTIES with the following prompts${RESET}\\n"
+  read -p "$(echo -e $WHITE)Enter your repo URL (repo where you want to version your applications):$(echo -e $RESET) " REPO_URL
+  read -p "$(echo -e $WHITE)Enter a folder name where you want this repo to be cloned (ex: staging):$(echo -e $RESET) " LOCAL_REPO
+  read -p "$(echo -e $WHITE)Enter a name of the branch you want to clone from your repo (Default is master):$(echo -e $RESET) " BRANCH_NAME
```
I wanted to leave this in for when we test internally too. Like I have a private gitlab repo where I host all the different changes of applications/configuration files/Jenkinsfile on different branches and I want the pipeline to use that branch. 

Leaving this out would mean there's one manual change to make when setting up all the files for testing. Unless we envision that if you are running this script, you only mean to use the master branch.

hankjtaylor replied:
I think its fair to have one that we use internally and we could distribute that to everyone and add to .gitignore, but since we have a clear recommended strategy that utilizes only master, I think we want to remove this. We also don't mention branching in the doc, so it may be confusing

samratjha96 commented on configurePipelineFiles.sh:
```diff
@@ -0,0 +1,226 @@
+      : # Do nothing if user has entry in the users.properties file
+    else
+      # TODO: Make a check that username/password isn't null
+      read -s -p "$(echo -e $WHITE)Enter the password for this user:$(echo -e $RESET) " TST_USR_PASSWORD
```
Are we planning on using sed in the Jenkinsfile to take care of this? That seems like the only other alternative

hankjtaylor replied:
Actually, I think this is fine. We specify under the security section that although the guide stores these passwords in plaintext, we recommend other configurations.

kpink224 reviewed (commented):
Looks pretty organized!

cobylu commented on configurePipelineFiles.sh:
```diff
@@ -0,0 +1,226 @@
+    cat $current_test
+    # TODO: Input validation
+    echo -e "${WHITE}Pick one of the following browsers:\\n
+      1. CHROME\\n
```
or make it so they can enter a number

cobylu commented on configurePipelineFiles.sh:
```diff
@@ -0,0 +1,226 @@
+      4. REMOTE_CHROME\\n"
+
+    read -p "(recommended REMOTE_FIREFOX or REMOTE_CHROME):$(echo -e $RESET) " TST_BROWSER
+    read -p "$(echo -e $WHITE)Enter the url for the Appian site you want to test (ending with /suite):$(echo -e $RESET) " TST_SITE_URL
```
I think it would be better for the user if we track that they entered an acceptance/integration site and offer that as a default?

cobylu commented on configurePipelineFiles.sh:
```diff
@@ -0,0 +1,226 @@
+|    Let's configure the ADM tools for use (AIM and AVM).   |
+|    The prerequisite is that you must have a AVM built     |
+|    application residing in the appian/applications folder |
+└-----------------------------------------------------------┘
```
How about something like "In addition to the configurations that are being set up now, you'll need to add and AVM build application to appian/applications folder"
 saying that the application is a prerequisite might be slightly ambiguous as to whether they need that before they proceed with the script...

Commit:
Addressing CR changes

samratjha96 commented:
@cobylu @hankjtaylor Ready for final round of CR. Check second commit for all the changes. 

1. Added more info to the messages displayed throughout the various stages
1. Changed unnecessary global variables to be local
1. Added parsing of the URL's to ensure they will all end up with /suite
1. Removed prompts for branching and the local repo path
1. Offer user the chance to select previously defined test and staging URL's for their integration and acceptance test configuration respectively
1. Make user specify a number when selecting among the browser options (continuous prompting until user selects a valid option)
1. Change the example user/password pair in users.properties to be more generic

hankjtaylor commented:
one minor comment about the auto-resizing, but other than that, LGTM!!

cobylu commented:
LGTM 🍣

samratjha96 commented:
Merging 🚀
